python/dgl/graphbolt/itemset.py

```diff
@@ -177,8 +177,8 @@ def __getitem__(self, index: Union[int, slice, Iterable[int]]):
                         f"{type(self).__name__} index out of range."
                     )
                 return torch.tensor(index, dtype=dtype)
-            elif isinstance(index, Iterable):
-                return torch.tensor(index, dtype=dtype)
+            elif isinstance(index, torch.Tensor):
+                return index.to(dtype)
```

Comment:
line 185 should be updated accordingly as well.

Reply:
Done.